docs/source/configuration/slots.md

```diff
@@ -0,0 +1,260 @@
+---
+myst:
+  html_meta:
+    "description": "Slots are insertion points in the Volto rendering tree structure."
+    "property=og:description": "Slots are insertion points in the Volto rendering tree structure."
+    "property=og:title": "Slots"
+    "keywords": "Volto, Plone, frontend, React, configuration, slots, viewlets"
+---
+
+# Slots
+
+Slots are insertion points in the Volto rendering tree structure.
+This concept is inherited from the Plone Classic UI {doc}`plone:classic-ui/viewlets`.
+
+
+## Anatomy
+
+Slots are named, and they can contain a list of different slot components.
+Slot components are also named, and they are registered in the configuration registry using a specific API for slots.
+
+The main trait of a slot component is that its renderer is controlled by a list of conditions called {term}`predicates`.
+Multiple slot components can be registered under the same name, as long as they have different predicates.
+
+The following tree structure diagram illustrates these concepts.
+
+```text
+Slot (`toolbar`)
+├── SlotComponent (`edit`)
+│   ├── predicates (only appear in `/de/about`)
+│   ├── predicates (only appear if the content type is either a `Document` or `News Item`)
+│   └── no predicates (default when all predicates return `false`)
+├── SlotComponent (`contents`)
+└── SlotComponent (`more`)
+```
+
+At the root of the tree, there is a slot named `toolbar`.
+It contains three slot components, named `edit`, `contents`, and `more`.
+`edit` contains two predicates and a default for when all of its predicates return `false`.
+
+Thus, when either the route is `/de/about` or the content type is either a `Document` or `News Item`, then the `edit` slot component would appear in the slot `about`.
+It would not display elsewhere.
+
+The order in which the components render is governed by the order in which they were registered.
+
+```{todo}
+You can change the order of the defined slot components for a different slot using the API. (pending)
+You can even delete the rendering of a registered slot component using the API. (pending)
+
+Slot (eg. `toolbar`)
+  - `edit`
+  - `contents`
+  - `more`
+```
+
+Volto renders slots using the `SlotRenderer` component.
+You can add insertion points in your code, as shown in the following example.
+
+```ts
+<SlotRenderer name="toolbar" content={content} />
```

Comment:
Why include Renderer in the name and not just call it Slot? It's obviously a renderer, since it's a component.

Reply:
(I don't feel strongly about this. I suppose it helps distinguish it from other kinds of objects related to slots.)